src/inference/AdvancedSMC.jl

```diff
@@ -110,8 +110,8 @@ function DynamicPPL.initialstep(
     # Reset the VarInfo.
     reset_num_produce!(vi)
     set_retained_vns_del_by_spl!(vi, spl)
-    resetlogp!(vi)
-    empty!(vi)
+    resetlogp!!(vi)
+    empty!!(vi)
```

Comment:
Does `empty!!` not call `resetlogp!!`? Maybe this would be reasonable? In any case, also here it is mandatory that the calls are actually mutating.

Reply:
Similar to the calls below.

Reply:
Same response as above: these samplers aren't compatible with immutable `VarInfo` :confused: